notebooks/data-sources/gcsweb-ci/build-logs/model_seldon.ipynb

```diff
@@ -0,0 +1,562 @@
+{
```

Comment:
<p>is this the right order for these commented lines? seems like the <code>delete</code>  portion should go before the upload file, in case users just want to uncomment the whole block</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/371/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Reply:
<p>On this note, in general I would prefer if control flow was not dictated by users needing to comment or un-comment sections of code manually. Is there any way you can check for the appropriate conditions instead?  </p>

Reply:
Yes, I agree, the comment and un-comment flow is quite confusing. In this case, the commented out code deletes a file on s3. I could uncomment it and leave it like that but it seems a little risky since it's a delete operation. I could remove it all together since the upload_file operation would anyway update the same file. The latter seems more reasonable to me. wdyt?

Reply:
> Is there any way you can check for the appropriate conditions instead?

Maybe something this would do the trick?
```python
ret = s3_resource.meta.client.list_objects_v2(Bucket=s3_bucket, Prefix='build-log-clustering/tfidf-kmeans/model.joblib"')
if ret['KeyCount'] > 0:
    # exists, dont upload again
else:
    # doesnt exist, go ahead with upload
```

But if the goal is to _always_ replace existing `model.joblib` with the current one, I think just doing `bucket.upload_file` without any checks might work, coz replaces existing object afaik :smile:

Reply:
yes, just remove the delete part